### ISSUE TITLE
Extendable Extensions

### ISSUE THREAD

**Issue by vscodenpa** [vscodenpa/testissues#30283]
I'd like to be able to add more features to an existing extension by extending it further, but without forking it and release an entirely new extension.

My reason for this is, coming from a Visual Studio extensibility background, we add items to context menus quite a bit to perform new actions. In the VSCode world, this might be the solution of project context menu. To get those views properly, you need the new C- DevKit extension to get the "solution explorer". To add new items and functionality to those menus, we need to be able to extend the C- DevKit (somehow).

In short, allow extensions to be extended by other extensions.
